linkml/utils/generator.py

```diff
@@ -76,6 +78,60 @@ def _resolved_metamodel(mergeimports):
     return metamodel
 
 
+T = TypeVar("T")
+
+
+def deprecated_fields(deprecated_map: dict[str, str]):
```

Comment:
Can you test this function directly in the `test_utils/test_generator.py` file to ensure that it's doing everything you expect it to?

Reply:
As mentioned [here](https://github.com/linkml/linkml/pull/2836#discussion_r2452999263), I think that this function should be in `linkml.utils.deprecation`. I will have to check how I can test it, but then in `test_utils/test_deprecation.py`.